src/DeVinci_frontend/components/ChatBox.svelte

```diff
@@ -41,6 +43,27 @@
 		return { update: scroll }
 	};
 
+  async function showNewChat() {
+    if ($activeChatGlobal !== null) {
+      $activeChatGlobal = null;
+    } else {
+      $activeChatGlobal = false;
+    }
+    if ($chatModelGlobal) {
+      try {
+        await $chatModelGlobal.interruptGenerate();
+      } catch (error) {
+        console.error("Error stopping the answer generation on starting new chat ", error);
+      }
+    }
+    messages = [];
+    newMessageText = '';
+    messageGenerationInProgress = false;
+    if (location.hash !== "#/") {
```

Comment:
this shouldn't be included in the function as we're already in this location and shouldn't change routes

Reply:
got it. simplified version in the commit below.

Reply:
looks good, thanks